### DISCUSSION TITLE
Guidance for writing your proposal

### DISCUSSION THREAD

**Discussion by danyx23** [owid/owid-grapher#1318] [Announcements]
Hi all!

Thank you very much for your interest in contributing to our codebase as part of this year's GSoC! We wanted to collect some information about the process to make sure that this process is as smooth as possible for everyone involved.

As you surely know, the big next step for you is to write a proposal for your work. We want to help you write the best application you can, and are happy to review and give feedback on draft proposals that you send to us. If we receive a draft proposal by end-of-day Apr 12 (one week before due date) then we should be able to give you feedback, but we encourage you to share a draft with us even sooner. The best format to use is a Google Doc or a PDF. Please note that you will still have to submit the final proposal to GSoC yourself, we will only provide feedback.

## Where to start

If you have not already done so, there are 3 steps that we would like you to take if you are interested in applying with us:

1. **Introduce yourself.** Make sure you have sent a first message introducing yourself to [gsoc@ourworldindata.org](mailto:gsoc@ourworldindata.org) so that we can get in contact with you when necessary. Sharing a CV helps us get a better idea of your skills and background but is not required.
2. **Read the ideas.** Read through the individual project ideas again and check [discussions page here on github](https://github.com/owid/owid-grapher/discussions). On the discussion pages we will answer general questions and add background information that is relevant to the specific projects so it is a good idea to check the discussion pages of the project(s) you are interested in regularly.
3. **Try our codebase.** Check out our Grapher repository and follow the steps to build the codebase and run the tests and be able to browse the codebase. This will be useful for investigating our codebase and finding relevant parts for your project that you can then mention in your proposal.

## How we will rank proposals

We will have to rank the proposals by how promising we find them and then Google will decide how many of the proposals we ranked will be greenlighted. This means that there is a higher chance for you to be selected for GSoC if you write a high quality proposal. Google has some [general guidelines](https://google.github.io/gsocguides/student/writing-a-proposal) for writing a proposal as well as [two](https://google.github.io/gsocguides/student/proposal-example-1) [examples](https://google.github.io/gsocguides/student/proposal-example-2).

A high quality proposal meets the following criteria:

1. **Demonstrated understanding.** The work you propose to do is relevant for our project and closely aligned with one of the ideas we outlined. It shows that you have understood what we need and are able to describe the problem you are working on in your own words and to outline the steps you intend to take as you work on it.
2. **The necessary skills.** You highlight which skills you think are needed and give us an indication that you command these skills or give a realistic outlook of what you intend to familiarize yourself with towards the beginning of the project.
3. **Communication.** You stay in a conversation with your mentor while you write your proposal and incorporate feedback but at the same time you are mindful of the time of your mentor.

## Mentoring

Initially, [Daniel Bachler](mailto:daniel.bachler@ourworldindata.org) will be the mentor for everyone while the process gets started. Other mentors might get involved during the proposal writing phase as we calibrate to the amount of incoming proposals. You can email him drafts of your proposals as you go (i.e. you can already send in drafts before the official proposal timeline starts).

## Some final thoughts

A quick word on our team at Our World In Data: This is the first time we are part of GSoC. We are very excited to be working with and mentoring students but we haven’t done this before. Please bear with us if we sometimes take a few days to answer, but feel free to let us know if we can improve an aspect of how we are trying to manage this process. Also, please feel free to get in touch if you have any remaining questions.

We’re excited to see your proposal! 

The Our World In Data developers team

**Comment by MahmoudHamdy02** [comment]
Hello, is there a recommended number of pages for the proposal?

**Reply by danyx23** [reply]
Hi @MahmoudHamdy02! Not really but most of the close to finished drafts I have seen so far fell in the 4-8 pages range to give you a rough ballpark range. But both shorter and longer proposals are possible!
